tests/mapper/test_graph_generator.py

```diff
@@ -0,0 +1,138 @@
+        example_graph_random(feature_size=4, n_nodes=5),
+    ]
+
+    graphs_mixed = [
+        example_graph_random(feature_size=2, n_nodes=6),
+        example_hin_1(is_directed=False),
+    ]
+
+    def test_generator_init(self):
+        generator = GraphGenerator(graphs=self.graphs)
+        assert len(generator.graphs) == len(self.graphs)
+
+    def test_generator_init_different_feature_numbers(self):
+        with pytest.raises(ValueError):
+            generator = GraphGenerator(graphs=self.graphs_diff_num_features)
+
+    def test_generator_init_nx_graph(self):
+        with pytest.raises(TypeError):
+            generator = GraphGenerator(graphs=self.graphs_nx)
+
+    def test_generator_init_hin(self):
+        with pytest.raises(ValueError):
+            generator = GraphGenerator(graphs=self.graphs_mixed)
+
+    def test_generator_flow_invalid_batch_size(self):
+        with pytest.raises(ValueError):
+            GraphGenerator(graphs=self.graphs).flow(graph_ilocs=[0], batch_size=-1)
+
+        with pytest.raises(TypeError):
+            GraphGenerator(graphs=self.graphs).flow(graph_ilocs=[0], batch_size=2.0)
+
+        with pytest.raises(ValueError):
+            GraphGenerator(graphs=self.graphs).flow(graph_ilocs=[0], batch_size=0)
+
+    def test_generator_flow_incorrect_targets(self):
+
+        generator = GraphGenerator(graphs=self.graphs)
+
+        with pytest.raises(ValueError):
+            generator.flow(graph_ilocs=[0, 1], targets=np.array([0]))
+
+        with pytest.raises(TypeError):
+            generator.flow(graph_ilocs=[0, 1], targets=1)
+
+    def test_generator_flow_no_targets(self):
+
+        generator = GraphGenerator(graphs=self.graphs)
+
+        seq = generator.flow(graph_ilocs=[0, 1, 2], batch_size=2)
+        assert isinstance(seq, GraphSequence)
+
+        assert len(seq) == 2  # two batches
+
+        # The first batch should be size 2 and the second batch size 1
+        batch_0 = seq[0]
+        assert batch_0[0][0].shape[0] == 2
+        assert batch_0[0][1].shape[0] == 2
+        assert batch_0[0][2].shape[0] == 2
+        assert batch_0[1] is None
+
+        batch_1 = seq[1]
+        assert batch_1[0][0].shape[0] == 1
+        assert batch_1[0][1].shape[0] == 1
+        assert batch_1[0][2].shape[0] == 1
+        assert batch_1[1] is None
+
+    def test_generator_flow_check_padding(self):
+
+        generator = GraphGenerator(graphs=self.graphs)
+
+        seq = generator.flow(graph_ilocs=[0, 2], batch_size=2)
+        assert isinstance(seq, GraphSequence)
+
+        assert len(seq) == 1
+
+        # The largest graph has 6 nodes vs 3 for the smallest one.
+        # Check that the data matrices have the correct size 6
+        batch = seq[0]
+
+        assert batch[0][0].shape[1] == 6
+        assert batch[0][1].shape[1] == 6
+        assert batch[0][2].shape[1] == 6
+
+        for mask in batch[0][1]:
+            assert np.sum(mask) == 6 or np.sum(mask) == 3
```

Comment:
Given we're passing in the graph ilocs in a fixed order, I think we know which mask is which and so can check them directly?

```suggestion
        assert np.sum(batch[0][1][0]) == 6
        assert np.sum(batch[0][1][1]) == 3
```

Reply:
Currently, this order cannot be guaranteed because when calling `flow` a `GraphSequence` object is created and the last statement in `GraphSequence.__init__` is to call `on_epoch_end` that shuffles the graph indexes. So, when the single batch is requested in the test, the order of the graphs within the batch is unknown. Maybe when #1043 is solved we can revisit this test.

Reply:
Ah, right 👍